### ISSUE TITLE
Changing how `split` command determines filenames

### ISSUE THREAD

**Issue by harivyasi** [mekkyz/ymlio#14]
At present, using the `split` command meaningfully **requires** that keys in the parent YAML file have an extension i.e.
```yaml
my_yaml_file.yml:
  key1: dict
  key2: str
```
produces a YAML file called `my_yaml_file.yml`. This then fails if the file is supplied from an external resource, e.g. is a valid `docker-compose` file.

This issue suggests a specification that changes this requirement and makes the tool more flexible. The specification is as follows:
1. Extension `.yml` is automatically appended to the key that is being exported as a file. So, the following will result in files `my_yaml_file1.yml` and `my_yaml_file2.yml`.
```yaml
my_yaml_file1:
  key1: dict
my_yaml_file2:
  key2: str
```  
2. If it is required that a given extension is used, the following format for key can be used. Following would create four files, namely `my_text_file1.txt`, `my_file_with_no_extension`, `.dotfile` and `my_yaml_file3.yml`.
```yaml
./my_text_file1.txt:
./my_file_with_no_extension:
my_yaml_file3:
  key3: float
./.dotfile:
```
3. The same can be extended to encode folders by using a trailing slash. The following should create two folders `my_folder` and `my_second_folder`, each containing a subfolder called `subfolder`.
```yaml
./my_folder/:
./my_folder/subfolder/:
./my_second_folder/subfolder/:
```
4. A combination of all of the above is also possible using the following notation:
```yaml
./my_third_folder/my_yaml_file4:
  key4: int
./my_fourth_folder/./my_file_without_extension
```

A suggestion would be to implement 1, 2, and 3 first with the possibility of implementing 4 if required.

**Comment by harivyasi** [comment]
@ptrxyz, your thoughts please.

**Comment by ptrxyz** [comment]
Hm, i do not see the reason why this breaks compose files.
Can you ELI5 this for my simple mind?

**Comment by harivyasi** [comment]
The current behavior means we can't split a single `docker-compose.yml` into multiple YAML files (that have an extension in the filename) because they have headers like `services` and `networks` and not `services.yml` and `networks.yml`.

If this an expected behavior then I am overthinking the problem (as I am prone to).
